### ISSUE TITLE
Failed to decode images transcoded from JPG to JXL

### ISSUE THREAD

**Issue by EugeneVert** [tirr-c/jxl-oxide#14]
**Describe the bug**
The following error message is shown when decoding transcoded images:
```bash
ERROR jxl_bitstream::reader: Out-of-order jxlp box found: expected 1, got 2147483649
thread 'main' panicked at 'failed to load frames: Frame(Bitstream(Io(Custom { kind: InvalidData, error: "Out-of-order jxlp box found" })))', crates/jxl-dec/src/main.rs:364:10
```

**To Reproduce**
```bash
convert -size 256x256 canvas:white 'example.jpg'
cjxl 'example.jpg' 'example.jxl'
jxl-dec 'example.jxl'
```

**Expected behavior**
Transcoded images from JPG to JXL successfully decoded

**Comment by tirr-c** [comment]
I've pushed the possible fix to the head (a0c36ce), does it fix the problem?

**Comment by EugeneVert** [comment]
Yes, it seems to be fixed, thanks!
